src/v2/view-builder/views/DeviceChallengePollView.js

```diff
@@ -2,25 +2,36 @@
 import { _, $, loc } from 'okta';
 import BaseView from '../internals/BaseView';
 import BaseForm from '../internals/BaseForm';
+import BaseFooter from '../internals//BaseFooter';
 import Logger from '../../../util/Logger';
 
-const request = (url, method = 'GET', body = null, timeout = 1000) => {
-  return $.ajax({
+const request = (opts) => {
+  let url;
+  let method = 'GET';
+  let body = null;
+  let timeout = null;
+  if (typeof opts === 'string') {
```

Comment:
This is over design IMO. the parameter could be either Object or multiple variables all the time.

Reply:
When is opts just a string that is equal to the api url ? For those cases can we just pass in an object instead `{ url : 'some url' }` and avoid this check ?